File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hm1 = {}
        for i in s:
            if i not in hm1:
                hm1[i] = 0
            hm1[i] += 1
        hm2 = {}
        for i in t:
            if i not in hm2:
                hm2[i] = 0
            hm2[i] += 1
        return hm1==hm2

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        
        if(s.length()!=t.length()){
            return false;
        }

        HashMap<Character, Integer> hm1 = new HashMap<>();

        for(int i=0;i<s.length();i++){
            if(!hm1.containsKey(s.charAt(i))){
                hm1.put(s.charAt(i), 1);
            }
            else{
                hm1.put(s.charAt(i), hm1.get(s.charAt(i))+1);
            }
        }
        HashMap<Character, Integer> hm2 = new HashMap<>();
        for(int j=0;j<t.length();j++){
            if(!hm2.containsKey(t.charAt(j))){
                hm2.put(t.charAt(j), 1);
            }
            else{
                hm2.put(t.charAt(j), hm2.get(t.charAt(j))+1);
            }
        }
        System.out.println(hm1+ "   -> "+hm2);
        
        // for(int i=0;i<s.length();i++){
        //     if ((hm1.get(s.charAt(i)) != hm2.get(s.charAt(i)))){
        //         return false;
        //     }
        // }
        // return true;
        
        return hm1.equals(hm2);


    }
}